pkg/virtualfs/virtualfs.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package virtualfs
+
+import (
+	"os"
+
+	"github.com/kopia/kopia/fs"
+)
+
+// NewDirectory returns new mock directory.
+func NewDirectory(rootName string) *Directory {
+	return &Directory{
+		dirEntry: dirEntry{
+			name: rootName,
```

Comment:
This should validate / sanity check `rootName` to not contain a path separator for instance.